v2/pkg/additional/local_stored_collector.go

```diff
@@ -116,14 +112,23 @@ func isImageByDigest(imgRef string) bool {
 	return strings.Contains(imgRef, "@")
 }
 
-func imageName(imgRef string) string {
-	var imageName string
-	imgSplit := strings.Split(imgRef, "@")
-	if len(imgSplit) > 1 {
-		imageName = imgSplit[0]
+func imagePathComponent(imgRef string) (string, error) {
```

Comment:
mini suggestion: imagePathComponent -> imagePath or imagePathComponents? 
This is because we are not returning 1 (singular) pathComponent, but the whole path without the domain (https://github.com/containers/image/blob/main/docker/reference/reference.go)